ext/phar/zip.c

```diff
@@ -161,6 +161,18 @@ static void phar_zip_u2d_time(time_t time, char *dtime, char *ddate) /* {{{ */
 }
 /* }}} */
 
+static char *phar_find_eocd(char *s, size_t n)
+{
+	char *p;
+
+	for (p = s + n - 1; p >= s; p--) {
+		if (*p == 'P' && ((p - s) + sizeof(phar_zip_dir_end) <= n && !memcmp(p + 1, "K\5\6", 3))) {
+			return p;
+		}
+	}
+	return NULL;
+}
```

Comment:
Maybe use zend_memnrstr?

Reply:
Unfortunately, `zend_memnrstr()` is static.

Reply:
It's declared in a header so that shouldn't be a problem?

Reply:
Ah, right! Thanks.